wiki/Publishing-guidelines/Writing-functional-specs.md

```diff
@@ -0,0 +1,383 @@
+
+### Data model
+At a low level every single one of our components starts here, and deviates as necessary. Here is a link to the [[Data model]].
+
+<br /><br />
+
+## Understanding the template
+
+The functional spec template is just a set of common tools that I find commonly used across components, and compiled into a single file/template for easy reference and better consistency. All of the sections in this template do not always have to be used, and while I would ask if we could avoid it the fact that we have this template doesn't mean it's the only way or even best way to present a particular component. Use it as a jumping off point, and try your best to stay consistent with it. With all things functional specs related collaboration is important, so we can discuss changing things when reviewing if necessary.
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  here is a link to the functional specs template: [[fs template]]
+
+<br />
+
+### Meta data
+
+The meta data is used for organizational purposes to help the reader find what they are looking for while also providing the team with language and breakdown of how a component is formed. Some of these items are auto generated, so there is no effort on your part to update while others are manual. 
+
+#### Table of contents
+The table of contents in the template is auto generated. As long as you are using proper markdown headings it should find them just fine, and list the page outline up at the top of the page. So make sure to include the table of contents section template in your component wiki page.
+
+#### Dependencies
+The dependency list is one of the most important aspects of the functional specs. It communicates to the developer, and creates almost this written agreement between design and development what components should be used when building a component out. 
+
+###### How to add a dependency
+
+1. Find the dependency list in your component's template.
+2. Add a new list item
+3. Link to the component using the Wiki linking strategy.
+
+<details>
+  <summary><strong>Example</strong></summary><br />
+
+```html
+- [[Component or wiki page name 1]]
+- [[Component or wiki page name 2]]
+- [[Component or wiki page name 3]]
+```
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Notice the double brackets with the file name `[[File name]]` in-between. This is different from a markdown link `[text](url)`.
+
+<br />
+</details><br /><br />
+
+#### Used by
+
+By properly linking dependencies we will also run an automation script to collect all the dependencies and create a used by list in addition. Used by components once populated communicates to the reader where this component is used.
```

Comment:
@photodow I'm a little unsure what we are telling the reader to do here. Is there something they need to do in the template? When I first read this, I thought the list was autogenerated based on the automation script you mention but then I read again and wasn't sure. Thanks.

Reply:
It's similar to the table of contents. There is a section in the template they need to make sure is included if they want to have a used by section. After that it just a matter of making sure the component is properly listed as a dependency in other components.

Reply:
Not perfect, but let me know if this makes anymore sense...
> Used by components communicates to the reader where this component is used in other components. As long as you include this, and properly link to it in other component's dependencies then this list will create itself.